Docker compose

winged opened the pull request:
NOTE: This depends on #2 so please review & merge that first

Commit:
Introduce Docker-Compose

Add psycopg2 as dependency, so we cna run postgres without further mess

oxivanisher reviewed (commented):
Looking good. I will implement a .env.example file to move the configuration out of the docker compose file after merging.

Also the volumes options are missing in the compose file.

Commit:
Dockerfile: shuffle order for faster builds

Only copy what's really needed, and don't "collectstatic" during build.
Also reorder the steps a bit, so when we rebuild the image, not
everything needs to be done again and again.

Commit:
data / storage directories for compose

Commit:
Boot command: Wait for DB before migrating

Also, put everything in the right order (migrate before creating
superuser, for instance, ...)

Commit:
Nginx config (WIP)

Commit:
WIP: uwsgi / nginx setup in composer.

Not working yet, but just backing up current state

Commit:
Move static files into app-specific subdirectory

Commit:
Rename "boot" management command into "init_modelreg"

This no longer "boots" the app, but merely prepares it. Starting
uwsgi or "runserver" is easier from outside, and easier to debug.

Commit:
Docker: use boot shell script instead of django mgmt cmd

Commit:
Docker-Compose: correct volume mount points and permissions

Commit:
Ignore storage in Git

Commit:
Base HTML: NO empty line at start of document

Commit:
Minor cosmetics & cleanups

* Remove STATICFILES_DIRS, as it's not needed anymore (our static files
are within the modelreg app).
* Set STATIC_ROOT in dev env

Commit:
Update / Integrate font-awesome

Commit:
nginx: Include mime types and other standard config options

This is required since we overwrite nginx's default config file, which
normally contains those.  YAY, static content works now!

Commit:
Docker-Compose: use modelreg image from dockerhub

Commit:
Add dockerignore for a cleaner build
